src/transformers/trainer.py

```diff
@@ -419,6 +419,12 @@ def __init__(
             raise ValueError(
                 f"You have set `args.eval_strategy` to {args.eval_strategy} but you didn't pass an `eval_dataset` to `Trainer`. Either set `args.eval_strategy` to `no` or pass an `eval_dataset`. "
             )
+        if args.save_strategy == SaveStrategy.BEST or args.load_best_model_at_end:
```

Comment:
@seanswyi Q: does this change mean that this logic:

https://huggingface.co/docs/transformers/main/main_classes/trainer#transformers.Seq2SeqTrainingArguments.metric_for_best_model

> ... Will default to "loss" if unspecified and load_best_model_at_end=True (to use the evaluation loss).

is broken now?

Reply:
I haven't tested it yet, but it seems like it is. The docs probably need an update.

Reply:
@shcheklein Yeah, I don't think that there's any case where `metric_for_best_model` default to loss.